velox/vector/arrow/Bridge.cpp

```diff
@@ -87,6 +137,46 @@ void exportFlatVector(const VectorPtr& vector, ArrowArray& arrowArray) {
   }
 }
 
+// Returns the Arrow C data interface format type for a given Velox type.
+const char* exportArrowFormatStr(const TypePtr& type) {
+  switch (type->kind()) {
+    // Scalar types.
+    case TypeKind::BOOLEAN:
+      return "b"; // boolean
+    case TypeKind::TINYINT:
+      return "c"; // int8
+    case TypeKind::SMALLINT:
+      return "s"; // int16
+    case TypeKind::INTEGER:
+      return "i"; // int32
+    case TypeKind::BIGINT:
+      return "l"; // int64
+    case TypeKind::REAL:
+      return "f"; // float32
+    case TypeKind::DOUBLE:
+      return "g"; // float64
+    case TypeKind::VARCHAR:
+      return "u"; // utf-8 string
+    case TypeKind::VARBINARY:
+      return "z"; // binary
+    case TypeKind::TIMESTAMP:
+      // TODO: need to figure out how we'll map this since in Velox we currently
+      // store timestamps as two int64s (epoch in sec and nanos).
+      return "ttn"; // time64 [nanoseconds]
+
+    // Complex/nested types.
+    case TypeKind::ARRAY:
+      return "+L"; // large list
```

Comment:
shall we use list here? -- large list is for data > 2**31 elements: https://arrow.apache.org/docs/python/generated/pyarrow.large_list.html

Reply:
good question. I think we need to use large list here because our size vector holds int64_t (unless we want to copy it to a narrower vector).